sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -556,6 +556,16 @@ case class StringSpace(child: Expression)
     UTF8String.fromBytes(spaces)
   }
 
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    nullSafeCodeGen(ctx, ev, (length) => {
+      val spaces = ctx.freshName("spaces")
+      s"""
+        byte[] $spaces = new byte[($length < 0) ? 0 : $length];
```

Comment:
can we move this into UTF8String, and just call that function both in the interpreted mode and in codegen?